src/main/java/seedu/task/logic/parser/FindCommandParser.java

```diff
@@ -19,15 +30,47 @@ public class FindCommandParser implements Parser<FindCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_DESCRIPTION, PREFIX_TAG);
+
+        if (areTooManyPrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_DESCRIPTION)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        }
+
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            return new FindCommand(new NameContainsKeywordsPredicate(argMultimap.getValue(PREFIX_NAME).get()));
+        }
+        if (argMultimap.getValue(PREFIX_DESCRIPTION).isPresent()) {
+            return new FindCommand(new DescContainsKeywordsPredicate(argMultimap.getValue(PREFIX_DESCRIPTION).get()));
         }
+        if (parseTagsForFind(argMultimap.getAllValues(PREFIX_TAG)).isPresent()) {
+            return new FindCommand(new TagsContainsKeywordsPredicate(argMultimap.getAllValues(PREFIX_TAG)));
+        }
```

Comment:
Would it be neater to put <name/desc/tags>ContainsKeywords under an abstract class in a package?
Might be a personal preference, but I think this can be further abstracted out by using dynamic binding.
Good use of streams and java classes :)